DuckDuckGo/TabViewController.swift

```diff
@@ -349,14 +352,18 @@ class TabViewController: UIViewController {
         
         if consumeCookies {
             consumeCookiesThenLoadRequest(request)
-        } else if let request = request {
-            if let url = request.url {
-                linkProtection.getCleanURL(from: url,
-                                           onStartExtracting: { showProgressIndicator() },
-                                           onFinishExtracting: { },
-                                           completion: { [weak self] cleanURL in self?.load(urlRequest: .userInitiated(cleanURL)) })
-            } else {
-                load(urlRequest: request)
+        } else if let url = request?.url {
+            var loadingStopped = false
+            linkProtection.getCleanURL(from: url) { [weak self] in
+                if loadedByParentTab {
+                    loadingStopped = true
+                    self?.webView.stopLoading()
+                }
+                showProgressIndicator()
+            } onFinishExtracting: {
+            } completion: { [weak self] cleanURL in
+                guard url != cleanURL || !loadedByParentTab || loadingStopped else { return }
```

Comment:
What is purpose for this guard?
Does it even makes sense, as it seems that both `loadedByParentTab` and `loadingStopped` will always have same values.

Reply:
the check here passes if either loading has stopped or if the tab is NOT loaded by its parent, done so not to interrupt an ongoing popup request loading if possible